### PR TITLE
Use a torrent file to spread the load.

### PR DESCRIPTION
**Pull request by stylemistake**
I am directly referencing the torrent file from Issue #39 because downloading using a torrent file is faster than using a magnet.

Enjoy a free data hosting. One side effect is that it's going to be a pain to update with new models.

:+1: 

**P.S.** I have only tested the download part, I have not tested the whole thing.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
-    wget "${DOWNLOAD_URL}/${MODEL_VERSION}/${MODEL_NAME}.meta" > /dev/null
-    wget "${DOWNLOAD_URL}/${MODEL_VERSION}/vocab.bpe" > /dev/null
+    echo "Downloading AIDungeon2 Model... (this may take a few minutes)"
+    mkdir -p "${MODELS_DIRECTORY}"
+    cd "${MODELS_DIRECTORY}"
+    mkdir "${MODEL_VERSION}"
+    apt-get install aria2 unzip > /dev/null
+    wget "${MODEL_TORRENT_URL}"
+    unzip "${MODEL_TORRENT_BASENAME}"
+    aria2c -x 16 -s 32 "${MODEL_TORRENT_BASENAME%.*}"
```
Comment (steamp0rt) [inline install.sh]:
```suggestion
    aria2c -x 16 -s 32 --seed-time=0 "${MODEL_TORRENT_BASENAME%.*}"
```

**Comment by stylemistake** [comment]
So, I have just tested this end to end on Colab (and fixed the issue when it continues to seed indefinitely).

It works.

**Comment by Akababa** [comment]
Can you see if you can mount this google drive shared folder? If it works it should be faster since there's no need to copy data: https://drive.google.com/drive/folders/1XiDD2BD8vLZaJxZpCrNYjscpvnD3EYrP

thanks!!

**Comment by stylemistake** [comment]
@nickwalton PR is ready to merge.

@Akababa This is out of scope of this PR. We should merge something which works right now, and we can add Google Drive as a web seed later.

**Comment by Akababa** [comment]
@stylemistake Sure, sorry for hijacking

**Comment by nickwalton** [comment]
Merged. Do we think this will work on Google colab too? Or more just for people for who want to experiment locally.

**Comment by Couls** [comment]
testing google colab now, the torrent gets as far as downloading but its VERY slow 
image

**Comment by nickwalton** [comment]
Yeah that's what I was worried about

**Comment by stylemistake** [comment]
Yeah, I guess we need more seed power to back it up, because Colab instances don't exhange it very well. This is what my torrent client reports:

image

**Comment by Portaluke** [comment]
Is there a way I can increase the efficiency of seeding on my end? I know my computer and network is capable of more, I just don't know what settings to mess with (if any) or if there's a more efficient torrent.

**Comment by Wildwestmars** [comment]
Only took 10 minutes for me, I made to help download the magnet.
I'm sure people are simply glad to be able to play this.

As a simpleton who can't code, I thank you all.

**Comment by MCterra10** [comment]
imdoingmypart

**Comment by stylemistake** [comment]
If you want to seed more efficiently, *increase maximum peer count to 500 or more*.

And this PR might improve situation a bit by also extending peer count to 500, so that it has a higher chance to land on an efficient seeder and make download speed more consistent. https://github.com/nickwalton/AIDungeon/pull/46

**Comment by Portaluke** [comment]
uTorrent_pUiv1guuJW
Hopefully this will do the trick, I'm heading off for the night and I'll leave this running to try and keep the servers moving. Good luck.

**Comment by edausq** [comment]
Adding my torrent in my seedbox, I hope it helps!

**Pull request merged**